src/index/VocabularyOnDisk.cpp

```diff
@@ -48,6 +48,34 @@ std::optional<string> VocabularyOnDisk::operator[](uint64_t idx) const {
   return result;
 }
 
+// _____________________________________________________________________________
+ad_utility::ConsumeratorImpl<std::string_view>
+VocabularyOnDisk::getWordWriterImpl(std::string outFileName) {
+  _file.open(outFileName.c_str(), "w");
+  ad_utility::MmapVector<IndexAndOffset> idsAndOffsets(
+      outFileName + _offsetSuffix, ad_utility::CreateTag{});
+  uint64_t currentOffset = 0;
+  std::optional<uint64_t> previousId = std::nullopt;
+  uint64_t currentIndex = 0;
+  while (co_await ad_utility::valueWasPushedTag) {
+    const auto& word = co_await ad_utility::nextValueTag;
```

Comment:
Find better names for the tags. And do we really need two `co_await`s here? Most natural would be something like `for (const auto& word = co_await nextWordTag) { ... }`.